src/components/Buzz/BuzzPurchase/BuzzTypeSelector.tsx

```diff
@@ -0,0 +1,127 @@
+import { Button, Group, Stack, Text } from '@mantine/core';
+import { IconBolt } from '@tabler/icons-react';
+import React from 'react';
+import { useBuzzCurrencyConfig } from '~/components/Currency/useCurrencyConfig';
+import type { PurchasableBuzzType } from '~/server/schema/buzz.schema';
+
+export interface BuzzTypeSelectorProps {
+  onSelect: (type: PurchasableBuzzType) => void;
+  onCancel?: () => void;
+  title?: string | null;
+  description?: React.ReactNode | null;
+  greenButton?: {
+    text?: string;
+    description?: React.ReactNode;
+  };
+  redButton?: {
+    text?: string;
+    description?: React.ReactNode;
+  };
+}
+
+export function BuzzTypeSelector({
+  onSelect,
+  onCancel,
+  title = 'Choose which Buzz to buy',
+  description = (
+    <>
+      We offer multiple types of Buzz for flexibility and control.
+      <br />
+      Pick the option that fits your needs—whether you want to keep things safe for work or unlock
+      the platform&apos;s full creative potential. Each type is tailored for different content and
+      payment preferences.
+    </>
+  ),
+  greenButton = {
+    text: 'Green Buzz',
+    description: (
+      <>
+        Can be purchased using <b>credit cards</b>.<br />
+        Can <b>only</b> be used to generate <b>safe for work</b> content.
```

Comment:
I know it's very nitpick, but let's go with strong instead of b
```suggestion
        Can be purchased using <strong>credit cards</strong>.<br />
        Can <strong>only</strong> be used to generate <strong>safe for work</strong> content.
```